packages/x-tree-view/src/useTreeItem2/useTreeItem2.ts

```diff
@@ -70,20 +73,44 @@ export const useTreeItem2 = <
         return;
       }
 
+      const rootElement = instance.getItemDOMElement(itemId);
+
+      // Don't blur the root when switching to editing mode
+      if (
+        status.editing ||
+        ((event.relatedTarget && (event.relatedTarget as HTMLElement))?.dataset?.element ===
```

Comment:
Nitpick: This is a slightly fragile check, but I'm not sure if we can make it better. 🙈

Reply:
We struggle to find a better approach, but this is clearly not ideal

Reply:
Yes, I hate this one. 🙈